clippy_lints/src/methods/mod.rs

```diff
@@ -1670,6 +1683,25 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
             }
         }
     }
+
+    fn check_trait_item(&mut self, cx: &LateContext<'tcx>, item: &'tcx TraitItem<'_>) {
+        if_chain! {
+            if item.ident.name == sym!(new);
```

Comment:
```suggestion
            if !in_external_macro(cx.tcx.sess, item.span);
            if item.ident.name == sym!(new);
```
Re-reading my previous comment I see now I was not clear, sorry about that. I think your idea of checking if we're inside an external macro was good, since that code can't be modified by the user. 

I wanted to say that *additionally* checking if the code is from expansion seems unnecessary, as it would not lint in cases like a local macro generating a trait definition, which can be changed by the user.

Reply:
Haha :smile: my bad, I read your previous comment too fast